crates/sui-core/src/authority.rs

```diff
@@ -422,6 +424,102 @@ impl AuthorityState {
+        consensus_output: &narwhal_consensus::ConsensusOutput,
+        transaction: SerializedTransaction,
+        total_batches: usize,
+        total_transactions: usize,
+    ) -> Result<(), SuiError> {
+        // Compute the next expected indices. Those will be persisted upon transaction execution
+        // and are only used for crash-recovery.
+        execution_indices.next(total_batches, total_transactions);
+
+        // The consensus simply orders bytes, so we first need to deserialize the transaction.
+        // If the deserialization fail it is safe to ignore the transaction since all correct
+        // clients will do the same. Remember that a bad authority or client may input random
+        // bytes to the consensus.
+        let (result, _outcome) = match bincode::deserialize::<
+            <AuthorityState as SingleExecutionState>::Transaction,
+        >(&transaction)
+        {
+            Err(e) => {
+                let error = narwhal_executor::SubscriberError::ClientExecutionError(format!(
+                    "Failed to deserialize transaction: {e}"
+                ));
+                // There is always a chance that the fault lies with our deserialization.
+                debug!("{error}");
+                (Ok(()), Err(error))
+            }
+            Ok(transaction) => {
+                // Execute the transaction. Note that the executor will need to choose whether to discard
+                // transactions from previous epochs by itself.
+                let result = self
+                    .handle_consensus_transaction(
+                        consensus_output,
+                        execution_indices.clone(),
+                        transaction,
+                    )
+                    .await;
+
+                match result {
+                    Ok(outcome) => (Ok(()), Ok(outcome)),
+                    Err(error) => match narwhal_executor::SubscriberError::from(error.clone()) {
+                        // We may want to log the errors that are the user's fault (i.e., that are neither
+                        // our fault or the fault of consensus) for debug purposes. It is safe to continue
+                        // by ignoring those transactions since all honest subscribers will do the same.
+                        non_fatal @ narwhal_executor::SubscriberError::ClientExecutionError(_) => {
+                            debug!("{non_fatal}");
+                            (Ok(()), Err(non_fatal))
+                        }
+                        // We must take special care to errors that are our fault, such as storage errors.
+                        // We may be the only authority experiencing it, and thus cannot continue to process
+                        // transactions until the problem is fixed.
+                        fatal => (Err(error), Err(fatal)),
+                    },
+                }
+            }
+        };
+        result
```

Comment:
these two methods are mostly borrowed from `SingleExecutor<State>`.

Reply:
Ok I think I understand how you implemented it. I think the original idea was to re-use the structure `SingleExecutor` in executor::Core, but what you have done works as well. 

tx_output is the sending end of this channel; https://github.com/MystenLabs/sui/blob/d027c4f9ce292450c0eddacf50b9012e7b5a54e0/crates/sui-core/src/authority_server.rs#L176

Reply:
added!